go/arrow/array/array.go

```diff
@@ -175,8 +175,8 @@ func init() {
 		arrow.LARGE_STRING:            func(data arrow.ArrayData) arrow.Array { return NewLargeStringData(data) },
 		arrow.LARGE_BINARY:            func(data arrow.ArrayData) arrow.Array { return NewLargeBinaryData(data) },
 		arrow.LARGE_LIST:              func(data arrow.ArrayData) arrow.Array { return NewLargeListData(data) },
-		arrow.INTERVAL:                func(data arrow.ArrayData) arrow.Array { return NewIntervalData(data) },
```

Comment:
I assume getting rid of interval is an intentional unrelated change and not a mistake?

Reply:
yes. If you look at `datatype.go` you can see there's a comment there pointing out that `INTERVAL` as an enum value is deprecated in favor of the individual interval enums (`INTERVAL_MONTH_DAY_NANO`, etc...) So adding the new `RUN_LENGTH_ENCODED` type seemed like a good time to clean that out.